### PR TITLE
Support named lwaftr instances and update ps program

### PR DESCRIPTION
**Pull request by tsyesika**
This adds support for the lwaftr to have a name, if one is specified in the YANG configuration then the name is claimed when the lwaftr starts.

This also changes the behaviour of the new `ps` command. The behaviour, now by default, will display the name if one exists and if not fallback to displaying the PID. This behaviour can be changed by using the `-p` or `--pid` flag which changes it back to the previous default, displaying all process by PID.

### CODE REVIEW

File: src/program/lwaftr/tests/data/icmp_on_fail.conf [hunk 1]
```diff
@@ -1,3 +1,4 @@
+name lwaftr1;
```
Comment (wingo) [inline src/program/lwaftr/tests/data/icmp_on_fail.conf]: I think this will lead to spurious test failures, as this file is used by many tests.  We wouldn't want to prevent multiple people from running "lwaftr check" at once.

File: src/program/ps/ps.lua [hunk 2]
```diff
@@ -14,26 +15,49 @@ end
...
+   function opt.p (arg) preferpid = true end
+   args = lib.dogetopt(args, opt, "h:p", {help='h', pid='p'})
    if #args ~= 0 then usage(1) end
+   return preferpid
+end
+
+local function appname_resolver()
+    local apps = {}
+    for name, pid in pairs(app.enumerate_named_programs()) do
+        apps[pid] = name
```
Comment (wingo) [inline src/program/ps/ps.lua]: Let's not use the name "app" here, as it also means the components of a flow network within snabb.  I suggest "instances", but many things can work

File: src/lib/yang/snabb-softwire-v1.yang [hunk 3]
```diff
@@ -19,6 +19,14 @@ module snabb-softwire-v1 {
     description
      "Configuration for Snabb lwaftr.";
 
+     leaf name {
```
Comment (wingo) [inline src/lib/yang/snabb-softwire-v1.yang]: Consider the case of wanting to run a number of Snabb applications on the command line with the same configuration (binding table etc) but different NICs.  Probably you want different names there too.  As PCI addresses for NICs are passed on the command line, I think users will also want to pass names on the command line.  Dunno.  Also if we include a name here in the YANG schema then we'd want "snabb config set /name foo" to work, and I think that maybe doesn't make sense.  WDYT?  We could certainly have name as state data of course, though that runs into other problems.

In short I suspect that passing the name on the command line is the most useful way to work.  WDYT?

File: src/program/ps/ps.lua (hunk 2 above)
Comment (kbara) [reply src/program/ps/ps.lua]: +1

File: src/program/lwaftr/setup.lua [hunk 4]
```diff
@@ -100,6 +101,10 @@ function lwaftr_app(c, conf)
    set_preprocessors(c, preprocessing_apps_v6, "lwaftr.v6")
    set_postprocessors(c, "lwaftr.v6", postprocessing_apps_v6)
    set_postprocessors(c, "lwaftr.v4", postprocessing_apps_v4)
+
+   if conf.name then
```
Comment (wingo) [inline src/program/lwaftr/setup.lua]: Would you mind moving this out to the code that handles --name?  I know it's not centralized but i would like to preserve the idea that the "conf" is a simple function of the config file and nothing else.
